### ISSUE TITLE
QUESTION:

### ISSUE THREAD

**Issue by MaenMax** [AlexPikalov/cdrs#261]
How to make Cassandra IP address/domain name configurable? 

let node = NodeTcpConfigBuilder::new("127.0.0.1:9042", NoneAuthenticator {}).build();

Above code works. But, whenever I add a variable instead of constant, Rust compiler complains that the variable does not live enough. Could you please help on this. 

Thanks
Maen

**Comment by MaenMax** [comment]
Any answer to this ?

**Comment by AlexPikalov** [comment]
Hi Maen,
Please provide a code snippet that reproduces your issue

**Comment by ArtRand** [comment]
I noticed something similar. 
This works: 
```rust
        // CASSANDRA_CONTACTS is initialized with lazy_static! (pulls from environment)
        if CASSANDRA_CONTACTS.is_empty() {
            return Err("failed to get CASSANDRA_SEEDS".to_string());
        }
        let mut nodes = Vec::new();
        for contact_point in CASSANDRA_CONTACTS.iter() {
            println!("adding node {}", contact_point);
            let tcp = TransportTcp::new(contact_point);
            if tcp.is_ok() {
                let node = NodeTcpConfigBuilder::new(contact_point, NoneAuthenticator)
                    .build();
                nodes.push(node);
                println!("added node {}", contact_point);
            } else {
                println!("{} isn't a good one", contact_point)
            }
        }
        if nodes.len() == 0 {
            return Err("none of the contact points were good".to_string());
        }
        let cluster_config = ClusterTcpConfig(nodes);
        let lb = RoundRobinSync::new();
        let session = new_session(&cluster_config, lb).expect("session");
```
this does not
```rust
        // say cassandra_hosts is provided as a string
        let contact_points = cassandra_hosts.split(",").collect::<Vec<_>>(); 
        let mut nodes = Vec::new();
        for contact_point in contact_points {
            println!("adding node {}", contact_point);
            let tcp = TransportTcp::new(contact_point);
            if tcp.is_ok() {
                let node = NodeTcpConfigBuilder::new(contact_point, NoneAuthenticator)
                    .build();
                nodes.push(node);
                println!("added node {}", contact_point);
            } else {
                println!("{} isn't a good one", contact_point)
            }
        }
        if nodes.len() == 0 {
            return Err("none of the contact points were good".to_string());
        }
        let cluster_config = ClusterTcpConfig(nodes);
        let lb = RoundRobinSync::new();
        let session = new_session(&cluster_config, lb).expect("session");
```

Maybe because of the lifetime [here](https://github.com/AlexPikalov/cdrs/blob/master/src/cluster/session.rs#L119)? How do you imagine to setup a session with something other than a string literal? 

(Really like using the lib btw.)

**Comment by AlexPikalov** [comment]
Hi @ArtRand,

Thanks for the snippet! It helped me to reproduce the problem. After some analysis I've came to conclusion that lifetime requirement can be relaxed so static lifetime is not mandatory.

@MaenMax Thanks for reporting the problem

**Comment by ArtRand** [comment]
Works great. Thanks for the speedy fix!

**Comment by MaenMax** [comment]
Thanks folks ! I tried with the new merge and it worked.

**Comment by AlexPikalov** [comment]
Cool! I'll keep you posted about publishing new version in this thread.

**Comment by AlexPikalov** [comment]
Try to use [`StaticPasswordAuthenticator`](https://docs.rs/cdrs/2.1.0/cdrs/authenticators/struct.StaticPasswordAuthenticator.html) instead of `PasswordAuthenticator`. It doesn't have any livetimes so it must be much easier to use. It was published in `2.1.0`
